### DISCUSSION TITLE
Support for spread operator in middleware matcher

### DISCUSSION THREAD

**Discussion by SamRoehrich** [vercel/next.js#42458] [Ideas]
### Describe the feature you'd like to request

I would like to have support for the spread operator in the middlware matcher so that if I can use an array of paths instead of having to manually type them out.

Consider a use case like:
```
// middlware.js
const paths = {
    '/profile': {
        func: () => {}
    },
    '/settings': {
        func: () => {}
    }
}

export function middleware(req) {
    const pathname = req.nextUrl.pathname

    if(Object.keys(paths).includes(pathname) return paths[pathname].func()

    return NextResponse.next()
}

export const matcher = [...Object.keys(paths)]
```
Currently the above example throws this error in the console:
`Unsupported spread operator in the Array Expression at "config.matcher".`

### Describe the solution you'd like

Support for the spread operator in the matcher.

### Describe alternatives you've considered

Writing out paths.

**Comment by thescientist13** [comment]
Yes, this would be great.  Trying to transition a lot of old -> new routes for a redesign will be a bit tedious having to list them all out individually 😵 

edit: I would add though for redirects, if that is your use case, the [`redirects` feature in _next.config.js_](https://nextjs.org/docs/api-reference/next.config.js/redirects) is working quite well as an alternative approach, just trying it out right now.

**Comment by kaloyanBozhkov** [comment]
Would be cool to have this, ran into it myself today.

**Comment by raflymln** [comment]
+1 on this

**Comment by YahyaBagia** [comment]
+1 on this

**Comment by kevinlandsberg** [comment]
+1

**Comment by leochiu-a** [comment]
+1

**Comment by acron0** [comment]
+1

**Comment by PrashantIndurkar** [comment]
+1

**Comment by albertininm** [comment]
+1

**Comment by GBvanDam** [comment]
Would it maybe even be stronger to enable multiple middlewares in a file-system based nested manner? This way, not only the spread operator is added, but several more things?

https://github.com/vercel/next.js/discussions/69866#discussion-7160165

**Comment by GBvanDam** [comment]
Lee Robinson pointed towards experimental interceptors https://github.com/vercel/next.js/pull/70961

these might answer at least part of this discussion?

**Comment by andrew-braun** [comment]
+1

If we have to put all our middleware into one file, I feel like it should be easy for us to separate the functions into their own files and import the matchers along with the functions.
